toolchain/check/eval.cpp

```diff
@@ -424,6 +424,15 @@ static auto GetConstantValue(EvalContext& eval_context,
   return inst_id;
 }
 
+static auto GetConstantValue(EvalContext& eval_context,
+                             SemIR::TypeInstId inst_id, Phase* phase)
+    -> SemIR::TypeInstId {
+  // The input instruction is a TypeInstId, and eval does not change types, so
```

Comment:
```suggestion
  // The input instruction is a TypeInstId, and eval does not change concrete types, so
```
(Eval will replace symbolic types with their values in the context.)

Reply:
I was trying to say it can change the values but the types of those values is unchanged. So if its value has type TypeType before eval, it will also be after. It's tricky to say this in prose though.

Reply:
// The input instruction is a TypeInstId, and eval does not produce values of
  // different types, so the result is also a valid TypeInstId.

Would that be correct?

Reply:
That's not true in general. For example, if an instruction of type `T*` is handed to the evaluator and the eval context says that `T` is `i32`, then the type of the result would be `i32*`, not `T*`. But it is true that eval doesn't change *concrete* types, other than perhaps replacing them with `<error>`.

Reply:
Ah, I see. And `TypeType` is concrete, even if the value was not. ok!

Reply:
I made this a bit more verbose since it is leaning on the invariants of TypeInstId

  // The input instruction is a TypeInstId, and eval does not change concrete
  // types (like TypeType which TypeInstId implies), so the result is also a
  // valid TypeInstId.